src/packetizer/mod.rs

```diff
@@ -1,115 +1,108 @@
-use crate::error::Error;
-use crate::extension::abs_send_time_extension::*;
-use crate::header::*;
-use crate::packet::*;
-use crate::sequence::*;
+use crate::errors::RTPError;
+use crate::packet::Packet;
+use crate::sequence::Sequencer;
+use crate::{extension::abs_send_time_extension::*, header::Header};
 
-use std::io::{BufWriter, Read};
-use std::time::{Duration, SystemTime};
+use std::time::Duration;
 
-#[cfg(test)]
 mod packetizer_test;
 
 // Payloader payloads a byte array for use as rtp.Packet payloads
 pub trait Payloader {
-    fn payload<R: Read>(&self, mtu: isize, reader: &mut R) -> Result<Vec<Vec<u8>>, Error>;
+    fn payload(&self, mtu: u16, payload: &[u8]) -> Vec<Vec<u8>>;
 }
 
-// Packetizer packetizes a payload
-pub trait Packetizer {
-    fn packetize<R: Read, P: Payloader, S: Sequencer>(
-        &mut self,
-        reader: &mut R,
-        payloader: &mut P,
-        sequencer: &mut S,
-        samples: u32,
-    ) -> Result<Vec<Packet>, Error>;
+/// Packetizer packetizes a payload
+pub trait PacketizerInterface {
+    fn packetize(&mut self, payload: &mut [u8], samples: u32) -> Result<Vec<Packet>, RTPError>;
     fn enable_abs_send_time(&mut self, value: u8);
 }
 
-// Depacketizer depacketizes a RTP payload, removing any RTP specific data from the payload
+/// Depacketizer depacketizes a RTP payload, removing any RTP specific data from the payload
 pub trait Depacketizer {
-    fn depacketize<R: Read>(&mut self, reader: &mut R) -> Result<(), Error>;
+    fn unmarshal(&mut self, packet: &mut [u8]) -> Result<Vec<u8>, RTPError>;
 }
 
 pub type FnTimeGen = fn() -> Duration;
 
-struct PacketizerImpl {
-    mtu: isize,
+struct Packetizer {
+    pub mtu: u16,
+    pub payload_type: u8,
+    pub ssrc: u32,
+    pub payloader: Box<dyn Payloader>,
+    pub sequencer: Box<dyn Sequencer>,
+    pub timestamp: u32,
+    pub clock_rate: u32,
+    pub abs_send_time: u8, //http://www.webrtc.org/experiments/rtp-hdrext/abs-send-time
+    pub time_gen: Option<FnTimeGen>,
+}
+
+pub fn new_packetizer(
```

Comment:
it seems anti-idiomatic rust.

Reply:
I'll be leaving this as is, `new_packetizer` returns an interface, we do not want to expose `PacketizerImpl`

Reply:
@rainliu would naming it `packetizer` instead be okay, or do you have a better suggestion?